js/bit4you.js

```diff
@@ -0,0 +1,624 @@
+                    'quoteId': quoteId,
+                    'active': active,
+                    'spot': true,
+                    'margin': false,
+                    'percentage': true,
+                    'swap': false,
+                    'future': false,
+                    'option': false,
+                    'contract': false,
+                    'tierBased': false,
+                    'precision': precision,
+                    'taker': taker,
+                    'maker': maker,
+                    'settle': undefined,
+                    'settleId': undefined,
+                    'type': 'spot',
+                    'feeSide': 'quote',
+                    'linear': undefined,
+                    'inverse': undefined,
+                    'contractSize': undefined,
+                    'expiry': undefined,
+                    'expiryDatetime': undefined,
+                    'strike': undefined,
+                    'optionType': undefined,
+                    'limits': limits,
+                    'info': market,
+                });
+            }
+            return result;
+        } catch (error) {
+            return error;
+        }
+    }
+
+    async fetchCurrencies (params = {}) {
+        try {
+            const response = await this.publicGetMarketAssets ();
+            const result = {};
+            const assetsIds = Object.keys (response);
+            for (let i = 0; i < assetsIds.length; i++) {
+                const key = assetsIds[i];
+                const code = this.safeCurrencyCode (key);
+                const asset = response[key];
+                const name = this.safeString (asset, 'name');
+                const fee = this.safeNumber (asset, 'withdraw_fee') || 0;
+                const limits = {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': this.safeNumber (asset, 'min_withdraw') || 0,
+                        'max': this.safeNumber (asset, 'max_withdraw') || 0,
+                    },
+                };
+                result[code] = {
+                    'id': key.toLowerCase (),
+                    'code': code,
+                    'name': name,
+                    'active': true,
+                    'fee': fee,
+                    'precision': 8,
+                    'info': asset,
+                    'limits': limits,
+                };
+            }
+            return result;
+        } catch (error) {
+            return error;
+        }
+    }
+
+    async fetchTransactions (code = undefined, since = undefined, limit = undefined, params = {}) {
+        const request = {};
+        request.since = since;
+        request.limit = limit;
+        request.iso = code;
+        request.after_id = params.after_id;
+        request.before_id = params.before_id;
+        try {
```

Comment:
Usually, we use try-catch at this level for very specific cases only (inside handleErrors for instance), so I'd suggest you remove them